### PR TITLE
Build and push image with prebuilt network interfaces

### PR DESCRIPTION
**Pull request by domire8**
As suggested by @eeberhard a few days back, I add here a workflow to build and push an image with the prebuilt network interfaces libraries and its dependencies. This goes to develop, and afterwards I'd like to make a new (minor) release because the workflow has to be on main to work I think.

### CODE REVIEW

**Comment by domire8** [comment]
The workflow has been tested successfully on my fork btw

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+ARG CPPZMQ_VERSION=4.7.1
 RUN wget https://github.com/zeromq/cppzmq/archive/v${CPPZMQ_VERSION}.tar.gz -O cppzmq-${CPPZMQ_VERSION}.tar.gz
 RUN tar -xzf cppzmq-${CPPZMQ_VERSION}.tar.gz
 WORKDIR /tmp/cppzmq-${CPPZMQ_VERSION}
 RUN mkdir build && cd build && cmake .. -DCPPZMQ_BUILD_TESTS=OFF && make -j install
+
 WORKDIR /tmp
-RUN rm -rf cppzmq*
+ARG CONTROL_LIBRARIES_BRANCH=develop
+RUN git clone -b ${CONTROL_LIBRARIES_BRANCH} --single-branch https://github.com/epfl-lasa/control-libraries.git
```
Comment (buschbapti) [inline Dockerfile:15]:
```suggestion
RUN git clone -b ${CONTROL_LIBRARIES_BRANCH} --depth 1 https://github.com/epfl-lasa/control-libraries.git
```

`--depth 1` is a bit stronger than `--single-branch` as it also pull only the latest commit. This makes things even lighter which is good here. There is no need for both as depth assumes single branch.

**Review by buschbapti — APPROVED** [review]
This is good and will help the development on lightweight interfaces. Thanks for that.

**Review by eeberhard — APPROVED** [review]
Nice work implementing this idea, and thanks for testing it on a fork. It looks correct to me and should be a useful image for downstream projects such as the lightweight interfaces

**Pull request merged**